### ISSUE TITLE
Error with error.identity.user.primaryEmailNotSpecified

### ISSUE THREAD

**Issue by enricodeleo** [marcelo-ochoa/oci-swarm-cluster#5]
Hello there!

I followed the steps for `arm` branch but get stuck on this error. Is there any other action I should do? Thanks in advance!  

```
Error: 400-IdcsConversionError, Post request failed{"schemas":["urn:ietf:params:scim:api:messages:2.0:Error","urn:ietf:params:scim:api:oracle:idcs:extension:messages:Error"],"detail":"The primary email must be specified.","status":"400","urn:ietf:params:scim:api:oracle:idcs:extension:messages:Error":{"messageId":"error.identity.user.primaryEmailNotSpecified"}}
Suggestion: Please retry or contact support for help with service: Identity User
Documentation: https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/identity_user
API Reference: https://docs.oracle.com/iaas/api/#/en/identity/20160918/User/CreateUser
Request Target: POST https://identity.eu-milan-1.oci.oraclecloud.com/20160918/users
Provider version: 4.79.0, released on 2022-06-08.
Service: Identity User
Operation Name: CreateUser
OPC request ID: acd65a0eefe91be179364b94d85d92bc/EEC547AAD76A4045009ABF519620EFD3/6FA975E7A516F41B9CEA2A06E6B80DB7
  on users.tf line 5, in resource "oci_identity_user" "oci_user" 
   5: resource "oci_identity_user" "oci_user" {
```

**Comment by marcelo-ochoa** [comment]
Hi:
   As the documentation say:
https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/identity_user
```
resource "oci_identity_user" "test_user" {
    #Required
    compartment_id = var.tenancy_ocid
    description = var.user_description
    name = var.user_name

    #Optional
    defined_tags = {"Operations.CostCenter"= "42"}
    email = var.user_email
    freeform_tags = {"Department"= "Finance"}
}
```
  email is optional, but may be now OCI user even are not designed for login need an email set. Try adding a property email at line 10 of users.tf file, this users is created to allow S3 API access to Oracle Object Storage. It looks like:
![image](https://user-images.githubusercontent.com/16995718/173829117-4410b12c-1c9d-4388-bc53-6e26fd10de06.png)


  Best regards, Marcelo.

**Comment by enricodeleo** [comment]
Thank you! I can confirm that you need to add `user_email` as variable. It can also be worthy notice that the configuration fails if the target compartment is not the root one.

The deployment is now successful for me, but how can I update docker- compose in order to apply DNS records?

**Comment by marcelo-ochoa** [comment]
Thanks a lot Enrico for your feedback.
Once the cluster is deployed and you want to update docker-compose.yml, just logged on node1 and perform:
```bash
$ ssh -i ssh_priv.key opc@oci-swarm-ID-0
$ sudo -i
# vi docker-compose.yml
# source /root/swarm.env
# export $(cut -d= -f1 /root/swarm.env)
# docker stack deploy -c docker-compose.yml swarm
```
Once you have a DNS configured to your Load Balancer IP, Traefik will register LetsEncrypt certs for Portainer, when this is ready you will be able to log into Portainer and start administering other stacks using the browser.
Best regards, Marcelo.

**Comment by enricodeleo** [comment]
Thanks. I am experienced with Portainer but unfortunately I am not with Oracle cloud and Terraform. I am having troubles understanding how to connect. I cannot understand which private key is attached to the nodes. I did provide one for DB, not for nodes. (Cluster started via `terraform apply` since stacks didn't worked well).

§§ UPDATE §§

Found myself. The output saved in the terraform state file is truncated so I cannot retrieve the key used for compute nodes. I destroyed the stack and created again with `generate_public_ssh_key` to `false` and `public_ssh_key` with a key I created beforehand.
